src/ucs/profile/profile_defs.h

```diff
@@ -55,22 +66,42 @@ typedef enum {
  *    < ucs_profile_record_t > * ucs_profile_thread_header_t::num_records
  *
  * ] * ucs_profile_thread_header_t::num_threads
+ * <env variables string>
  */
 
 
+typedef struct {
+    uint64_t offset;
+    uint64_t size;
+} UCS_S_PACKED ucs_profile_block_t;
+
+
+#define UCS_PROFILE_WRITE_BLOCK(_fd, _profile_block, _func, ...) \
+    ({ \
+        ucs_status_t _status; \
+        _profile_block.offset = lseek(_fd, 0, SEEK_CUR); \
+        _status               = _func(_fd, __VA_ARGS__); \
+        _profile_block.size = lseek(_fd, 0, SEEK_CUR) - _profile_block.offset; \
+        _status; \
+    })
```

Comment:
1. can we move to C file?
2. can we use structures sizes calc instead of "lseek"?

Reply:
1. Yes
2. This macro solves the problem for future blocks/features of the profiling file. It might be a bit complex to calculate the size of the threads block because the thread's amount of records may vary, and the calculation depends on UCX_PROFILE_MODE as well. We can't even use padding without calculating the thread with max records.